apps/src/JsDebuggerUi.js

```diff
@@ -533,3 +568,38 @@ JsDebuggerUi.prototype.onStepOutButton = function() {
     this.updatePauseUiState();
   }
 };
+
+/**
+ * Refresh the watch variables.
+ * @private
+ */
+JsDebuggerUi.prototype.onWatchTimer = function() {
+  var jsInterpreter = this.jsInterpreter_;
+  if (jsInterpreter) {
+    var debugWatchDiv = this.getElement_('#debug-watch');
+    if (debugWatchDiv) {
+      while (debugWatchDiv.firstChild) {
+        debugWatchDiv.removeChild(debugWatchDiv.firstChild);
```

Comment:
Yes, the rendering is one of the more temporary pieces, but I'll update to reduce some duplication.